service/src/test/java/com/epam/rft/atsy/service/impl/StatesServiceImplTest.java

```diff
@@ -263,50 +281,53 @@ public void getCandidateApplicationsByCandidateIdShouldReturnAnEmptyList() {
-        then(statesRepository).should(times(AT_ONE_TIME)).findTopByApplicationEntityOrderByStateIndexDesc(applicationEntity);
+        then(candidateRepository).should().findOne(FIRST_ID);
+        then(applicationsRepository).should().findByCandidateEntity(firstCandidateEntity);
+        then(statesRepository).should(times(TWO_TIMES)).findTopByApplicationEntityOrderByStateIndexDesc(firstApplicationEntity);
     }
 
     @Test
     public void getCandidateApplicationsByCandidateShouldReturnAListWithThreeElements() {
         // Given
-        given(candidateRepository.findOne(FIRST_ID)).willReturn(candidateEntity);
-        given(applicationsRepository.findByCandidateEntity(candidateEntity)).willReturn(applicationEntityListWithThreeElements);
-        given(statesRepository.findTopByApplicationEntityOrderByStateIndexDesc(applicationEntity)).willReturn(stateEntity);
+        given(candidateRepository.findOne(FIRST_ID)).willReturn(firstCandidateEntity);
+        given(applicationsRepository.findByCandidateEntity(firstCandidateEntity)).willReturn(applicationEntityListWithThreeElements);
+        given(statesRepository.findTopByApplicationEntityOrderByStateIndexDesc(firstApplicationEntity)).willReturn(firstStateEntity);
+        given(statesRepository.findTopByApplicationEntityOrderByStateIndexDesc(secondApplicationEntity)).willReturn(secondStateEntity);
+        given(statesRepository.findTopByApplicationEntityOrderByStateIndexDesc(thirdApplicationEntity)).willReturn(thirdStateEntity);
 
         // When
         Collection<CandidateApplicationDTO> candidateApplicationDTOCollection = statesService.getCandidateApplicationsByCandidateId(FIRST_ID);
 
         // Then
-        assertCandidateApplicationDTOCollection(candidateApplicationDTOCollection, getExpectedCandidateApplicationDTOListFromApplicationEntityListAndStateEntity(applicationEntityListWithThreeElements, stateEntity));
+        assertCandidateApplicationDTOCollection(candidateApplicationDTOCollection, getExpectedCandidateApplicationDTOListFromApplicationEntityList(applicationEntityListWithThreeElements));
 
-
-        then(candidateRepository).should(times(AT_ONE_TIME)).findOne(FIRST_ID);
-        then(applicationsRepository).should(times(AT_ONE_TIME)).findByCandidateEntity(candidateEntity);
-        then(statesRepository).should(times(THREE_TIMES)).findTopByApplicationEntityOrderByStateIndexDesc(applicationEntity);
+        then(candidateRepository).should().findOne(FIRST_ID);
+        then(applicationsRepository).should().findByCandidateEntity(firstCandidateEntity);
+        then(statesRepository).should(times(TWO_TIMES)).findTopByApplicationEntityOrderByStateIndexDesc(firstApplicationEntity);
+        then(statesRepository).should(times(TWO_TIMES)).findTopByApplicationEntityOrderByStateIndexDesc(secondApplicationEntity);
+        then(statesRepository).should(times(TWO_TIMES)).findTopByApplicationEntityOrderByStateIndexDesc(thirdApplicationEntity);
```

Comment:
The StateServiceImpl executes a single stateRepository.findTopByApplicationEntityOrderByStateIndexDesc() for each application entities.

Please remove the usage of the mock in the getExpectedCandidateApplicationDTOList.... method.